### PR TITLE
4.12 Release Notes for P/Z

### PR DESCRIPTION
**Pull request by alishaIBM**
Jira for P/Z : 
https://issues.redhat.com/browse/MULTIARCH-3037

Draft for P/Z :
https://docs.google.com/document/d/1YC6_zrytSgWWBl6sjtMkKTV-12ev8JCL2SZfl1Jr3cU/edit#

Reviewers :
For P : Manoj Kumar, Luvella
for Z : Holger Wolf, Florian Leber

Preview :
For P : https://53464--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes.html#ocp-4-12-ibm-power
For Z : https://53464--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes.html#ocp-4-12-ibm-z

Branch : enterprise-4.12
@SNiemann15

### CODE REVIEW

**Comment by lsmcfadden** [comment]
@alishaIBM (and @manojnkumar FYI so you can correct me if needed)  In the **preview** and/or **doc draft** 

1. I see IPv6 listed under "_the following operators are supported_" - unsure if the PR commits it that way or not.  Do not think we call IPv6 support an operator?
2. I see FIPS cryptography listed both under notable enhancements AND under restrictions.  I infer this is because we are waiting 'to see' when this support comes in but intend to have it on only one list in the final copy?
3. There is some variance between p and z under the 'restrictions' section which I infer makes technical sense but just want to mention it so that we are sure.  Examples are CRC is in the Power list (but not the z list), Red Hat OpenShift on Local is on the z list (but not the Power list), and Persistent non-shared storage must be provisioned using local storage, like iSCSI, FC, or using LSO with DASD, FCP, or EDEV/FBA is on the z list (but not the Power list).

File: release_notes/ocp-4-12-release-notes.adoc [hunk 1]
```diff
@@ -143,9 +143,186 @@ In {product-title} 4.12, you can do the following from the *Helm* page:
...
+* Three-node cluster support
+* 4K Disk Support
+
+[discrete]
+==== Restrictions
+
+The following restrictions impact {product-title} on IBM Power:
+
+* Automatic repair of damaged machines with machine health checking
+* CodeReady Containers (CRC)
```
Comment (SNiemann15) [inline release_notes/ocp-4-12-release-notes.adoc:315]:
```suggestion
* {openshift-local-productname}
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 2]
```diff
@@ -143,9 +143,186 @@ In {product-title} 4.12, you can do the following from the *Helm* page:
...
+* Controlling overcommit and managing container density on nodes
+* FIPS cryptography
```
Comment (SNiemann15) [inline release_notes/ocp-4-12-release-notes.adoc:317]:
```suggestion

```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 3]
```diff
@@ -143,9 +143,186 @@ In {product-title} 4.12, you can do the following from the *Helm* page:
...
+[discrete]
+==== Supported features
+
+The following features are also supported on IBM Power:
+
+* Currently, the following Operators are supported:
+** Cluster Logging Operator
+** Compliance Operator
+** File Integrity Operator
+** IPv6
```
Comment (SNiemann15) [inline release_notes/ocp-4-12-release-notes.adoc:272]:
```suggestion

```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 4]
```diff
@@ -143,9 +143,186 @@ In {product-title} 4.12, you can do the following from the *Helm* page:
...
+** IPAM
+** IPVLAN
+* Alternate Authentication Provider
+* CSI Volumes
+** Cloning
+** Expansion
+** Snapshot
+* Encrypting data stored in etcd
+* Helm
+* Horizontal pod autoscaling
```
Comment (SNiemann15) [inline release_notes/ocp-4-12-release-notes.adoc:292]:
```suggestion
* Horizontal pod autoscaling
* IPv6
```

**Review by SNiemann15 — CHANGES_REQUESTED** [review]

**Comment by SNiemann15** [comment]
> @alishaIBM (and @manojnkumar FYI so you can correct me if needed) In the **preview** and/or **doc draft**
> 
> 1. I see IPv6 listed under "_the following operators are supported_" - unsure if the PR commits it that way or not.  Do not think we call IPv6 support an operator?
> 2. I see FIPS cryptography listed both under notable enhancements AND under restrictions.  I infer this is because we are waiting 'to see' when this support comes in but intend to have it on only one list in the final copy?
> 3. There is some variance between p and z under the 'restrictions' section which I infer makes technical sense but just want to mention it so that we are sure.  Examples are CRC is in the Power list (but not the z list), Red Hat OpenShift on Local is on the z list (but not the Power list), and Persistent non-shared storage must be provisioned using local storage, like iSCSI, FC, or using LSO with DASD, FCP, or EDEV/FBA is on the z list (but not the Power list).

@lsmcfadden Your correct with your findings I've added exact suggestions how to change. Regarding your comment about the Persistent non-shared storage entry we have for IBM Z. Is it exactly the same for Power and @alishaIBM  can just copy our entry or any variations?

File: release_notes/ocp-4-12-release-notes.adoc [hunk 5]
```diff
@@ -143,9 +143,186 @@ In {product-title} 4.12, you can do the following from the *Helm* page:
...
+* Scheduler profiles
+* Stream Control Transmission Protocol (SCTP)
+
+[discrete]
+==== Technology Preview features
+
+:FeatureName: Installing {op-system} using IBM Secure Execution
+include::snippets/technology-preview.adoc[]
+
+For installation instructions, see the following documentation:
```
Comment (SNiemann15) [inline release_notes/ocp-4-12-release-notes.adoc:172]:
```suggestion
For installation instructions, see the following documentation: 

* xref:../installing/installing_ibm_z/installing-ibm-z-kvm.html#installing-rhcos-using-ibm-secure-execution_installing-ibm-z-kvm[Installing {op-system} using IBM Secure Execution]
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 6]
```diff
@@ -143,9 +143,186 @@ In {product-title} 4.12, you can do the following from the *Helm* page:
...
+* Cron jobs
+* Descheduler
+* FIPS cryptography
+* IPv6
+* PodDisruptionBudget
+* Scheduler profiles
+* Stream Control Transmission Protocol (SCTP)
+
+[discrete]
+==== Technology Preview features
```
Comment (SNiemann15) [inline release_notes/ocp-4-12-release-notes.adoc:167]:
```suggestion
==== IBM Secure Execution (Technology Preview)

{product-title} now supports configuring {op-system-first} nodes for IBM Secure Execution on {ibmzProductName} and LinuxONE (s390x architecture) as a Technology Preview feature.
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 7]
```diff
@@ -143,9 +143,185 @@ In {product-title} 4.12, you can do the following from the *Helm* page:
...
+* xref:../installing/installing_ibm_z/installing-restricted-networks-ibm-z-kvm.adoc#installing-restricted-networks-ibm-z-kvm[Installing a cluster with RHEL KVM on IBM Z and LinuxONE in a restricted network]
+
+[discrete]
+==== Notable enhancements
+
+The following new features are supported on IBM Z and LinuxONE with {product-title} {product-version}:
+
+* Cron jobs
+* Descheduler
+* FIPS cryptography
```
Comment (SNiemann15) [inline release_notes/ocp-4-12-release-notes.adoc:160]:
```suggestion
//* FIPS cryptography
```
@alishaIBM In todays program call we decided to take FIPS cryptography out and readd through change mgmt after GA see here: https://docs.google.com/document/d/1lgW4fqzzPo4C2zBd81vcKBLtbRH7tDuBN8ODKwW8Idk/edit#heading=h.rq8qakhcqcki

File: release_notes/ocp-4-12-release-notes.adoc [hunk 8]
```diff
@@ -143,9 +143,185 @@ In {product-title} 4.12, you can do the following from the *Helm* page:
...
+
+[discrete]
+==== Notable enhancements
+
+The following new features are supported on IBM Power with {product-title} {product-version}:
+
+* Cloud controller manager for IBM Cloud
+* Cron jobs
+* Descheduler
+* FIPS cryptography
```
Comment (SNiemann15) [inline release_notes/ocp-4-12-release-notes.adoc:257]:
```suggestion
//* FIPS cryptography
```

File: release_notes/ocp-4-12-release-notes.adoc (hunk 7 above)
Comment (alishaIBM) [reply release_notes/ocp-4-12-release-notes.adoc:160]: Acknowledge

**Comment by Amrita42** [comment]
Nice work lgtm

**Pull request merged**